filebeat/harvester/reader/docker_json.go

```diff
@@ -31,6 +38,55 @@ func NewDockerJSON(r Reader, stream string) *DockerJSON {
 	}
 }
 
+// parseCRILog parses logs in CRI log format.
+// CRI log format example :
+// 2017-09-12T22:32:21.212861448Z stdout 2017-09-12 22:32:21.212 [INFO][88] table.go 710: Invalidating dataplane cache
+func parseCRILog(b []byte, msg *crioLog, message Message) (Message, error) {
+	log := string(b)
+	ts, err := time.Parse(time.RFC3339, strings.Fields(log)[0])
+	if err != nil {
+		return message, errors.Wrap(err, "parsing CRI timestamp")
+	}
+
+	stream := strings.Fields(log)[1]
```

Comment:
Calling strings.Fields only once should be more efficient, you can do it at the beginning of this method and use the resulting variable. It will probably be more efficient if you use `SplitN`, to as you don't want to split the whole message.

Also it would be nice to check the resulting array, to ensure it's size is correct, and avoid errors when accessing different fields

Reply:
Thanks for your comments @exekias !
I will use `SplitN` and check array len, if less than 3 then it'll be considered an invalid CRI log.

Reply:
I'm planning to test the code on a running cluster, I tried yesterday but had some issues while setting up the environment on my Windows system.